src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -590,19 +590,16 @@ internal int BytesToWrite
         internal SerialStream(string portName, int baudRate, Parity parity, int dataBits, StopBits stopBits, int readTimeout, int writeTimeout, Handshake handshake,
             bool dtrEnable, bool rtsEnable, bool discardNull, byte parityReplace)
         {
-            int flags = NativeMethods.FILE_FLAG_OVERLAPPED;
-
-            if ((portName == null) || !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase))
+            if ((portName == null) ||
+                !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase) || 
+                !uint.TryParse(portName.Substring(3), out uint portNumber))
+            {
                 throw new ArgumentException(SR.Arg_InvalidSerialPort, nameof(portName));
+            }
 
             // Error checking done in SerialPort.
 
-            SafeFileHandle tempHandle = Interop.Kernel32.CreateFileDefaultSecurity(
-                @"\\?\" + portName,
-                Interop.Kernel32.GenericOperations.GENERIC_READ | Interop.Kernel32.GenericOperations.GENERIC_WRITE,
-                0,              // comm devices must be opened w/exclusive-access
-                FileMode.Open,  // comm devices must use OPEN_EXISTING
-                flags);
+            SafeFileHandle tempHandle = OpenPort(portNumber);
```

Comment:
This is adding two extra string allocations for Win32 that weren't there before: one for `Substring(3)` check above (at least until #22403 is available) and another for the `portNumber.ToString()`.

Why not have `OpenPort` take a string, and do the `TryParse` check for Uap only? e.g. something like:

SerialStream.Win32.cs:

```c#
public SafeFileHandle OpenPort(string portName)
{
    return Interop.Kernel32.CreateFileDefaultSecurity(
        @"\\?\" + portName,
        ...
}
```

SerialStream.Uap.cs:

```c#
public SafeFileHandle OpenPort(string portName)
{
    if (!uint.TryParse(portName.Substring(3), out uint portNumber)
    {
        throw new ArgumentException(SR.Arg_InvalidSerialPort, nameof(portName));
    }

    return Interop.KernelBase.OpenCommPort(
        portNumber,
        ...
}
```